src/eckit/utils/RLE.h

```diff
@@ -14,19 +14,22 @@
 #ifndef eckit_RLE_h
 #define eckit_RLE_h
 
-
+#include <chrono>
 #include <iosfwd>
 
 //-----------------------------------------------------------------------------
 
 namespace eckit {
 
+using EncodingClock = std::chrono::steady_clock;
+
 //-----------------------------------------------------------------------------
 
 class Stream;
 
 template <class InputIterator, class OutputIterator>
-long long RLEencode2(InputIterator first, InputIterator last, OutputIterator result, long long maxloop);
+long long RLEencode2(InputIterator first, InputIterator last, OutputIterator result, long long maxloop,
+                     EncodingClock::duration timelimit = EncodingClock::duration::zero());
```

Comment:
I like the use of optional in the inner function, I think the interface would be nice if this was also used here, i.e. 
```suggestion
long long RLEencode2(InputIterator first, InputIterator last, OutputIterator result, long long maxloop,
                     const std::optional<EncodingClock::duration>& timelimit = std::nullopt);
```